src/Kit/System/Watchdog/_0test/_rawthread/rawthread.cpp

```diff
@@ -0,0 +1,335 @@
+*/
+#define SUPERVISOR_HEALTH_CHECK_MS 200
+
+/** Raw thread watchdog timeout.
+    This is shorter than the supervisor timeout to ensure the raw thread
+    failure is detected and handled appropriately.
+*/
+#define RAW_THREAD_WDOG_TIMEOUT_MS 400
+
+/** Time to run normally before triggering the stuck condition.
+    This gives the watchdog time to initialize and start running properly.
+*/
+#define NORMAL_RUN_DELAY_MS        2000
+
+/** How long the raw thread will be stuck.
+    This must exceed RAW_THREAD_WDOG_TIMEOUT_MS to trigger watchdog trip.
+*/
+#define STUCK_DURATION_MS          ( RAW_THREAD_WDOG_TIMEOUT_MS + 500 )
+
+//------------------------------------------------------------------------------
+// Raw Thread Implementation
+//------------------------------------------------------------------------------
+
+/// Counter for raw thread to track when to get stuck
+static volatile uint32_t rawThreadCounter_ = 0;
+
+/// Flag to trigger the stuck condition
+static volatile bool triggerRawThreadStuck_ = false;
+
+/// Runnable for the raw thread
+class RawThreadRunnable : public IRunnable
+{
+protected:
+    /// Watchdog monitor for this raw thread
+    RawThread m_wdog;
+
+public:
+    /** Constructor. Initializes the raw thread with watchdog monitoring.
+     */
+    RawThreadRunnable() noexcept
+        : m_wdog( RAW_THREAD_WDOG_TIMEOUT_MS )
+    {
+    }
+
+public:
+    /// Thread entry point
+    void entry() noexcept override
+    {
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Raw thread starting (wdog timeout: %u ms)\r\n",
+                              (unsigned)RAW_THREAD_WDOG_TIMEOUT_MS );
+
+        // Start watchdog monitoring for this thread
+        KIT_SYSTEM_WATCHDOG_START_RAWTHREAD( m_wdog );
+
+        // Run normally until we're told to get stuck
+        while ( !triggerRawThreadStuck_ )
+        {
+            // Do some work
+            Bsp_toggle_debug2();
+
+            // Kick the watchdog to indicate we're alive
+            KIT_SYSTEM_WATCHDOG_KICK_RAWTHREAD( m_wdog );
+
+            // Wait a bit (less than our watchdog timeout)
+            sleep( 100 );
+            rawThreadCounter_ += 100;
+        }
+
+        // Now get stuck in a busy loop WITHOUT kicking the watchdog
+        KIT_SYSTEM_TRACE_MSG( SECT_, "*** RAW THREAD STUCK - Entering busy loop ***\r\n" );
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Busy loop will last %u ms (thread timeout is %u ms)\r\n",
+                              STUCK_DURATION_MS, RAW_THREAD_WDOG_TIMEOUT_MS );
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Supervisor should detect timeout and trip watchdog\r\n" );
+
+        uint32_t startTime = HAL_GetTick();
+        while ( ( HAL_GetTick() - startTime ) < STUCK_DURATION_MS )
+        {
+            // Busy wait - watchdog will NOT be kicked
+            Bsp_toggle_debug2();
+            // Small delay to make LED visible
+            for ( volatile uint32_t i = 0; i < 100000; i++ )
+            {
+                __NOP();
+            }
```

Comment:
You can adjust/change the `50` constant to meet the "visible" requirement
```suggestion
            sleep(50);
```